Item adapter

murphymatt opened the pull request:
Created adapter and xml files for beer objects. Modified classes include beerAdapter and beerItem. Associated xml file created for list item. Will develop items for food and drink according to future modifications of beer item (and provided designs).

murphymatt committed:
Created ItemAdapter class

murphymatt committed:
Created ItemAdapter class

murphymatt committed:
Created ItemAdapter class

murphymatt committed:
added generic menu list item xml

murphymatt committed:
filled out textview

murphymatt committed:
added circleimageview to xml

murphymatt committed:
instantiated getItem class in ItemAdapter

murphymatt committed:
renamed MenuAdapter to BeerAdapter

murphymatt committed:
renamed menu_list_item to beer_list_item

murphymatt committed:
create folder for adapters

murphymatt committed:
Created additionally necessary adapter classes

murphymatt committed:
created menuItems directory

murphymatt committed:
Created BeerItem model

murphymatt committed:
created FoodItem model

murphymatt committed:
created DrinkItem model

murphymatt committed:
filled out BeerItem model

murphymatt committed:
created rating field with getter and setter

murphymatt committed:
removed rating field

murphymatt committed:
created price field

murphymatt committed:
started getImageResource method

murphymatt committed:
fixed getPrice method

murphymatt committed:
filled in DrinkItem class

murphymatt committed:
created general MenuItem model class

murphymatt committed:
Created MenuItem interface

murphymatt committed:
added getImageResource method

murphymatt committed:
Added method implementations

murphymatt committed:
filled out methods

murphymatt committed:
removed visibility modifiers

murphymatt committed:
filled out adapter and item classes

murphymatt committed:
added image view to xml

murphymatt committed:
reformatted xml

murphymatt committed:
reformatted xml

murphymatt committed:
filled beer adapter with beer data fields

LarryAsante commented on app/src/main/java/grinnell/appdev/edu/lyles/Adapters/BeerAdapter.java:
```diff
@@ -2,13 +2,17 @@
 
 import android.content.Context;
 import android.graphics.Picture;
+import android.hardware.camera2.params.StreamConfigurationMap;
```
Woah, what is this import doing? Wanna check and organize your imports?

LarryAsante commented on app/src/main/java/grinnell/appdev/edu/lyles/Adapters/BeerAdapter.java:
```diff
@@ -19,6 +23,8 @@
  * adapter contains image, and description of catelog item
  */
 public class BeerAdapter extends ArrayAdapter<BeerItem> {
+    private static final String TAG = ArrayAdapter.class.getSimpleName();
```
Using a tag is good in logging and testing things on your own. Unless it is particularly necessary, you shouldn't have a tag here. I like that you made it `static final`. Immediately, though, you don't need it :)

LarryAsante commented on app/src/main/java/grinnell/appdev/edu/lyles/Adapters/BeerAdapter.java:
```diff
@@ -36,14 +42,20 @@ public View getView(int pos, View convertView, ViewGroup parent) {
             convertView = LayoutInflater.from(this.getContext()).inflate(R.layout.beer_list_item, parent, false);
```
nit: Formatting. Please use proper level indentation.  See code guideline or my PR in Publications as seen here: https://github.com/GrinnellAppDev/Publications-Android/blob/develop/app/src/main/java/edu/grinnell/grinnell_publications_android/Activities/MainActivity.java#L105

LarryAsante commented on app/src/main/java/grinnell/appdev/edu/lyles/Adapters/BeerAdapter.java:
```diff
@@ -36,14 +42,20 @@ public View getView(int pos, View convertView, ViewGroup parent) {
-            // create references to view items
-            // final ImageView beerImage = convertView.findViewById(R.id.beerImage);
-        }
+            Log.d("beer", "Beer item is not null");
```
Please don't leave log statements in code pushed out into production. Log statements should only be on your local machine.

LarryAsante commented on app/src/main/java/grinnell/appdev/edu/lyles/Adapters/BeerAdapter.java:
```diff
@@ -36,14 +42,20 @@ public View getView(int pos, View convertView, ViewGroup parent) {
-            // final ImageView beerImage = convertView.findViewById(R.id.beerImage);
-        }
+            Log.d("beer", "Beer item is not null");
+            ImageView beerImage = (ImageView) convertView.findViewById(R.id.beer_image);
```
`beerImage` is assigned once and never changed. How about you make it `final` ?

LarryAsante commented on app/src/main/java/grinnell/appdev/edu/lyles/Adapters/BeerAdapter.java:
```diff
@@ -36,14 +42,20 @@ public View getView(int pos, View convertView, ViewGroup parent) {
+            // beerImage.setImageDrawable((drawable) beer.getImageResource());
 
-        
+            TextView  beerName  = (TextView)  convertView.findViewById(R.id.beer_name);
```
Same as `beerImage` above

LarryAsante commented on app/src/main/java/grinnell/appdev/edu/lyles/Adapters/BeerAdapter.java:
```diff
@@ -36,14 +42,20 @@ public View getView(int pos, View convertView, ViewGroup parent) {
+            TextView  beerName  = (TextView)  convertView.findViewById(R.id.beer_name);
+            beerName.setText(beer.getName());
+
+            TextView  beerType  = (TextView)  convertView.findViewById(R.id.beer_type);
```
Same as above

LarryAsante reviewed (changes requested):
Nice job! Please check the comments below

LarryAsante commented on app/src/main/java/grinnell/appdev/edu/lyles/menuItems/BeerItem.java:
```diff
@@ -1,16 +1,18 @@
     public BeerItem() {
         mName = null;
         mDescription = null;
+        mPrice = 0.00;
```
What is 0.00 ? Is it some magic number? Please don't use magic numbers. Btw, you should just leave this as a default constructor. All the fields will be at their default values.

LarryAsante commented on app/src/main/java/grinnell/appdev/edu/lyles/menuItems/DrinkItem.java:
```diff
@@ -5,4 +5,28 @@
+    private String mName;
+    private String mDescription;
+
+    public DrinkItem() {
```
Keep this as an empty constructor

LarryAsante commented on app/src/main/java/grinnell/appdev/edu/lyles/menuItems/DrinkItem.java:
```diff
@@ -5,4 +5,28 @@
 
 public class DrinkItem {
+
+    private String mName;
```
Nice! You're using the `m_` format. Keep that up!

LarryAsante commented on app/src/main/java/grinnell/appdev/edu/lyles/menuItems/BeerItem.java:
```diff
@@ -1,16 +1,18 @@
 
 /**
- * Model beer item. Contains
+ * Model beer item. Contains information on beer name and beer description
```
Good that you have the javadoc here. The first line of your comment should be what the class does. Please put all the explanation for the description on the next line

LarryAsante reviewed (changes requested):
Looking good - just some few changes and we'll be good to go!

LarryAsante commented on app/src/main/java/grinnell/appdev/edu/lyles/menuItems/MenuItem.java:
```diff
@@ -4,40 +4,18 @@
  */
 
-public class MenuItem {
+public interface MenuItem {
```
`MenuItem` is a class in the Android SDK. You probably want to call this something else. Good that you're using an Interface!

LarryAsante commented on app/src/main/java/grinnell/appdev/edu/lyles/menuItems/MenuItem.java:
```diff
@@ -4,40 +4,18 @@
  * Created by mattmurphy on 9/25/16.
```
Please improve the Javadoc

LarryAsante reviewed (changes requested):
Few comments :)

murphymatt committed:
remove unused imports, log statement, tag

murphymatt committed:
set adapter fields to final

murphymatt committed:
set fields to private, created constructor, reformatted empty getImageResources method

murphymatt committed:
updated comments for OrderItem class

LarryAsante commented:
@murphymatt : pull the new master branch,  make the changes suggested in this review and submit it for review. Good job, though!

murphymatt committed:
Added beer placeholder strings

murphymatt committed:
fixed merge issues

murphymatt committed:
merge changes

# Conflicts:
#	app/src/main/java/grinnell/appdev/edu/lyles/fragments/MenuFragment.java

murphymatt committed:
added image view and text views to beer resource file

murphymatt committed:
formatted textviews in relative layout

murphymatt committed:
equal spacing of text fields in beer_layout

murphymatt committed:
added star unselect imagebutton

murphymatt committed:
removed unused relative layout

murphymatt committed:
added padding to star

murphymatt committed:
added favorite star image

murphymatt committed:
changed id of star button

murphymatt committed:
added field to set image resource according to favorite status

murphymatt committed:
added favorite field and functions to beer model

murphymatt committed:
added onclick listener for beer button

murphymatt committed:
changed favorite set function
